src/sql/workbench/services/commandLine/common/commandLineService.ts

```diff
@@ -57,81 +60,71 @@ export class CommandLineService implements ICommandLineProcessing {
 		}
 	}
 
-	public _serviceBrand: any;
 	// We base our logic on the combination of (server, command) values.
 	// (serverName, commandName) => Connect object explorer and execute the command, passing the connection profile to the command. Do not load query editor.
 	// (null, commandName) => Launch the command with a null connection. If the command implementation needs a connection, it will need to create it.
 	// (serverName, null) => Connect object explorer and open a new query editor
 	// (null, null) => Prompt for a connection unless there are registered servers
-	public processCommandLine(args: ParsedArgs): Promise<void> {
-		let profile = undefined;
+	public async processCommandLine(args: ParsedArgs): Promise<void> {
```

Comment:
what does adding async to the prototype get us?

Reply:
i suppose it simplifies the code by enabling use of "await" instead of the manual resolve/reject calls

Reply:
Yes it allows use of await which streamlines the code quite a bit.